src/Dev/Deprecation.php

```diff
@@ -67,29 +67,46 @@ class Deprecation
      * must be available before this to avoid infinite loops.
      *
      * This will be overriden by the SS_DEPRECATION_ENABLED environment if present
+     *
+     * @internal - Marked as internal so this and other private static's are not treated as config
      */
-    protected static bool $is_enabled = false;
+    private static bool $is_enabled = false;
 
     /**
      * @var array
      * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
      */
     protected static $module_version_overrides = [];
 
-    protected static bool $inside_notice = false;
+    /**
+     * @internal
+     */
+    private static bool $inside_notice = false;
 
     /**
-     * Switched out by unit-testing to E_USER_NOTICE because E_USER_DEPRECATED is not
-     * caught by $this->expectDeprecated() by default
-     * https://github.com/laminas/laminas-di/pull/30#issuecomment-927585210
-     *
-     * @var int
+     * @var array
+     * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
      */
     public static $notice_level = E_USER_DEPRECATED;
 
+    /**
+     * Buffer of user_errors to be raised when enabled() is called
+     *
+     * This is used when setting deprecated config via yaml, before Deprecation::enable() has been called in _config.php
+     * Deprecated config set via yaml will only be shown in the browser when using ?flush=1
+     * It will not show in CLI when running dev/build flush=1
+     *
+     * @internal
+     */
+    private static array $user_error_message_buffer = [];
+
     public static function enable(): void
     {
         static::$is_enabled = true;
+        foreach (self::$user_error_message_buffer as $message) {
+            user_error($message, E_USER_DEPRECATED);
+        }
+        self::$user_error_message_buffer = [];
```

Comment:
Why do we send deprecation notices here instead of when calling `static::config()->get('some_deprecated_config')`?

Reply:
No, happens when we set them via yml - see code comment - https://github.com/silverstripe/silverstripe-framework/pull/10559/files#diff-c086526ebde3894cd8edfdde6b51e19939d0e4e288438e82ef562ac3b8533278R95

Reply:
Right. My questions is why? It seems like firing notices when trying to _get_ the values would be more valuable? Or both, perhaps - but _especially_ getting the values I would expect would be important.

e.g. I have functionality that depends on some config value. That value has been deprecated - so I need to know to stop relying on that value.

Reply:
In fact the acceptance criteria explicitly mentions getting the values as the time to throw notices:
> If a value for a deprecated configuration property is fetched, a deprecation notice is emitted.

Throwing notices when _setting_ values is noted as a possibly valuable item, but isn't the main point of the card.

Please make it so that when getting values, deprecations notices are fired.

Reply:
> please state that
Deprecation::notice() is thrown in the silverstripe/config PR that's linked in the parent issue - https://github.com/silverstripe/silverstripe-config/pull/73/files#diff-c836ffa3eb3ac044164ee747f070290bd6a9c5b0300eacd147bf79357ddf6689R121